crates/iceberg/src/spec/snapshot.rs

```diff
@@ -0,0 +1,343 @@
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub parent_snapshot_id: Option<i64>,
+    /// A monotonically increasing long that tracks the order of
+    /// changes to a table.
+    pub sequence_number: i64,
+    /// A timestamp when the snapshot was created, used for garbage
+    /// collection and table inspection
+    pub timestamp_ms: i64,
+    /// The location of a manifest list for this snapshot that
+    /// tracks manifest files with additional metadata.
+    pub manifest_list: String,
+    /// A string map that summarizes the snapshot changes, including operation.
+    pub summary: Summary,
+    /// ID of the table’s current schema when the snapshot was created.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub schema_id: Option<i64>,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "kebab-case")]
+/// A snapshot represents the state of a table at some time and is used to access the complete set of data files in the table.
+pub(crate) struct SnapshotV1 {
+    /// A unique long ID
+    pub snapshot_id: i64,
+    /// The snapshot ID of the snapshot’s parent.
+    /// Omitted for any snapshot with no parent
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub parent_snapshot_id: Option<i64>,
+    /// A timestamp when the snapshot was created, used for garbage
+    /// collection and table inspection
+    pub timestamp_ms: i64,
+    /// The location of a manifest list for this snapshot that
+    /// tracks manifest files with additional metadata.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub manifest_list: Option<String>,
+    /// A list of manifest file locations. Must be omitted if manifest-list is present
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub manifests: Option<Vec<String>>,
+    /// A string map that summarizes the snapshot changes, including operation.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub summary: Option<Summary>,
+    /// ID of the table’s current schema when the snapshot was created.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub schema_id: Option<i64>,
+}
+
+impl From<SnapshotV2> for Snapshot {
+    fn from(v2: SnapshotV2) -> Self {
+        Snapshot {
+            snapshot_id: v2.snapshot_id,
+            parent_snapshot_id: v2.parent_snapshot_id,
+            sequence_number: v2.sequence_number,
+            timestamp_ms: v2.timestamp_ms,
+            manifest_list: ManifestList::ManifestListFile(v2.manifest_list),
+            summary: v2.summary,
+            schema_id: v2.schema_id,
+        }
+    }
+}
+
+impl From<Snapshot> for SnapshotV2 {
+    fn from(v2: Snapshot) -> Self {
+        SnapshotV2 {
+            snapshot_id: v2.snapshot_id,
+            parent_snapshot_id: v2.parent_snapshot_id,
+            sequence_number: v2.sequence_number,
+            timestamp_ms: v2.timestamp_ms,
+            manifest_list: match v2.manifest_list {
```

Comment:
Hmm, in PyIceberg we don't check for the `manifests` field. cc @rdblue

Reply:
The spec says it's required in v2?
<img width="1044" alt="image" src="https://github.com/apache/iceberg-rust/assets/2771941/8a35a3de-3201-49f1-bc26-765e9fe81160">